### ISSUE TITLE
Dropdown displaced when using pinch zoom

### ISSUE THREAD

**Issue by ZhaofengWu** [select2/select2#5048]
This is reproducible in the [demo page](https://select2.org/getting-started/basic-usage). Click on the first example, and then zoom in. The dropdown will become displaced. Tested on macOS Sierra with Chrome.

![image](https://user-images.githubusercontent.com/11954789/30731632-17c0603e-9fa1-11e7-9e7c-893056174195.png)

**Comment by ZhaofengWu** [comment]
Looks similar to yet actually different from #4058.

**Comment by alexweissman** [comment]
I'm also using Sierra but I can't seem to replicate this.  Which version of Chrome are you using?

When I zoom in (using Cmd++), it seems to behave as expected:

![screen shot 2017-09-22 at 10 52 08 am](https://user-images.githubusercontent.com/5004534/30750353-469fac60-9f84-11e7-87bc-2e130074417e.png)

**Comment by ZhaofengWu** [comment]
I was zooming in by pinching on the trackpad. Cmd++ works fine for me too.

**Comment by ZhaofengWu** [comment]
Reproducible in Safari 11.0 (12604.1.38.1.7) as well.

![image](https://user-images.githubusercontent.com/11954789/30750874-31d66204-9fea-11e7-9b5d-f6ebdb98c0c9.png)

**Comment by alexweissman** [comment]
Interesting!  It would seem that in Chromium, pinch-zoom and browser zoom are two completely separate implementations 😱 

See this Chromium bug report: https://bugs.chromium.org/p/chromium/issues/detail?id=378826

**Comment by ZhaofengWu** [comment]
Hmm, but why the displacement in Safari? 

I also tried Firefox, which seems only have the browser zoom, hence not having this problem.

**Comment by alexweissman** [comment]
I'm not sure, but Safari might have a similar implementation of pinch-zoom.  It's not clear if this is a problem that Select2 should attempt to solve.  Is there any cross-browser compatible way to detect a pinch-zoom in Javascript?

**Comment by ZhaofengWu** [comment]
Not that I am aware of. Other similar libraries do not seem to have this problem though. E.g. [Selectize](https://selectize.github.io/selectize.js/) and [Chosen](https://harvesthq.github.io/chosen/). But it might be a difference in implementation.

**Comment by alexweissman** [comment]
Yeah, not really sure.  Select2 attaches the dropdown to the body (see https://select2.org/dropdown#dropdown-placement), so it's possible that pinch-zoom is doing something to the body element that is causing this problem.

Can you try using `dropdownParent` to attach to a different container element, and see if the problem persists?

**Comment by ZhaofengWu** [comment]
The problem disappears when I attach the dropdown to a different container. For example, the following code eliminates the problem:

```
$('.js-example-basic-single').select2({
  dropdownParent: $('#body-inner')
});
```

**Comment by alexweissman** [comment]
Alright, we can add this to the documentation!

**Comment by alexweissman** [comment]
See https://select2.org/troubleshooting/common-problems#the-dropdown-becomes-misaligned-displaced-when-using-pinch-zoom.

**Comment by ZhaofengWu** [comment]
It seems that it does not work with any arbitrary `div` that is more specific than `body`. I am not able to prove this point in the Select2 demo, but I can reproduce it in the [Bootstrap theme demo](https://select2.github.io/select2-bootstrap-theme/). The problem persists after running:

```
$("select").first().select2({
    dropdownParent: $(".jumbotron")
})
```

The dropdown UI becomes weird due to the loss of the theme, but the point still holds.
